modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/EmailAddressResourceTest.java

```diff
@@ -275,6 +285,29 @@ private long _getListTypeId(String listTypeId) {
 		return listType.getListTypeId();
 	}
 
+	private void _testDeletePrimaryEmailAddress() throws Exception {
+		EmailAddress emailAddress1 = randomEmailAddress();
+
+		emailAddress1.setPrimary(true);
+
+		emailAddress1 = _addEmailAddress(
+			emailAddress1, Contact.class.getName(), _user.getContactId(),
+			ListTypeConstants.CONTACT_EMAIL_ADDRESS);
+
+		Assert.assertTrue(emailAddress1.getPrimary());
+
+		EmailAddress emailAddress2 = testDeleteEmailAddress_addEmailAddress();
+
+		Assert.assertFalse(emailAddress2.getPrimary());
```

Comment:
Same idea here as with phone.

Reply:
Tbh, I would leave it there (just in case, that one line hurts noone)... I have the same logic in this case as you but Stefano pointed out something very similar in one of my previous PRs, so lets see what he says.

Reply:
@smottal ^

Reply:
My logic with this was a "make sure the email address(or other contact) was saved succesfully as intended. Also, in the end, this becames primary, so the change from non-primary to primary is asserted. But I'm not sure how "strict" Brian is with making assertions that are like 99% will be true.

Reply:
For me is fine like it is for this reason
1) we "show" that primary is false
2) we do something
3) we "show" that primary is true

Reply:
@brbalazs in your PR the scenario was different. You were setting the type equal to the default value because was not provided in the random*** (if I remember correctly)